File: 1759-count-number-of-homogenous-substrings/1759-count-number-of-homogenous-substrings.py
class Solution:
    def countHomogenous(self, s: str) -> int:
        result = 0
        count = 0
        mod = 10 ** 9 + 7
        
        for i, char in enumerate(s):
            if i == 0 or s[i-1] == char:
                count += 1
            else:
                count = 1
            
            result += count
        
        return result % mod
            

File: 1759-count-number-of-homogenous-substrings/1759-count-number-of-homogenous-substrings.js
/**
 * @param {string} s
 * @return {number}
 */
var countHomogenous = function(s) {
    let result = 0;
    let count = 0;
    let mod = 10 ** 9 + 7
    
    for(let i = 0; i < s.length; i++){
        
        if(i == 0 || s.charAt(i) == s.charAt(i-1)){
            count += 1;
        }
        else{
            count = 1;
        }
        
        result += count;
    }
    
    return result % mod;
};